### PR TITLE
DOC: tweak paragraph regarding cut and IntervalIndex

### PR DESCRIPTION
**Pull request by ghost**
I thought this paragraph could be improved, so I tried to.

### CODE REVIEW

File: doc/source/user_guide/advanced.rst [hunk 1]
```diff
@@ -965,21 +965,28 @@ If you select a label *contained* within an interval, this will also select the
...
 
 .. ipython:: python
 
    c = pd.cut(range(4), bins=2)
    c
    c.categories
 
-Furthermore, ``IntervalIndex`` allows one to bin *other* data with these same
-bins, with ``NaN`` representing a missing value similar to other dtypes.
+:func:`cut` also accepts an ``IntervalIndex`` for its `bins` argument, which enables
```
Comment (WillAyd) [inline doc/source/user_guide/advanced.rst]: Double backticks for anything that is not a role here

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/advanced.rst [hunk 2]
```diff
@@ -965,21 +965,29 @@ If you select a label *contained* within an interval, this will also select the
...
+the ``bins`` argument in subsequent calls to :func:`cut`, supplying new data which
+will be binned into the same bins.
 
 .. ipython:: python
 
    pd.cut([0, 3, 5, 1], bins=c.categories)
 
+When :func:`cut`  is passed an ``IntervalIndex`` for the `bins` argument, The ``Interval``
+values in the ``IntervalIndex`` define the bins and any value which falls outside all
+bins will be assigned a ``NaN`` value.
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/advanced.rst]: I would keep it to the last part here (the former is already explained before the example), like "Values that fall outside the passed Intervals will be assigned a `NaN` value".

File: doc/source/user_guide/advanced.rst (hunk 1 above)
Comment (pilkibun) [reply doc/source/user_guide/advanced.rst]: fixed.

File: doc/source/user_guide/advanced.rst (hunk 2 above)
Comment (pilkibun) [reply doc/source/user_guide/advanced.rst]: You're right. Updated.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@pilkibun Thanks!

**Pull request merged**